libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationManager.java

```diff
@@ -0,0 +1,115 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.athenz.common.server.notification;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.yahoo.athenz.common.server.notification.NotificationServiceConstants.NOTIFICATION_PROP_SERVICE_FACTORY_CLASS;
+
+public class NotificationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationManager.class);
+
+    private NotificationService notificationService;
+    private ScheduledExecutorService scheduledExecutor;
+    private List<NotificationTask> notificationTasks;
+
+    public NotificationManager(List<NotificationTask> notificationTasks) {
+        this.notificationTasks = notificationTasks;
+        String notificationServiceFactoryClass = System.getProperty(NOTIFICATION_PROP_SERVICE_FACTORY_CLASS);
+        if (notificationServiceFactoryClass != null) {
+            NotificationServiceFactory notificationServiceFactory;
+            try {
+                notificationServiceFactory = (NotificationServiceFactory) Class.forName(notificationServiceFactoryClass).newInstance();
+                notificationService = notificationServiceFactory.create();
+                init();
+            } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                LOGGER.error("Invalid NotificationServiceFactory class: " + notificationServiceFactoryClass + " error: " + e.getMessage());
+            }
+        }
+    }
+
+    private void init() {
+        if (enableScheduledNotifications()) {
+            scheduledExecutor = Executors.newScheduledThreadPool(1);
+            scheduledExecutor.scheduleAtFixedRate(new NotificationsSender(), 0, 1, TimeUnit.DAYS);
+        }
+    }
+
+    public NotificationManager(final NotificationServiceFactory notificationServiceFactory, List<NotificationTask> notificationTasks) {
+        this.notificationTasks = notificationTasks;
+        notificationService = notificationServiceFactory.create();
+        init();
+    }
+
+    public void shutdown() {
+        if (scheduledExecutor != null) {
+            scheduledExecutor.shutdownNow();
+        }
+    }
+
+    public void sendNotifications(List<Notification> notifications) {
+        if (isNotificationFeatureAvailable()) {
+            notifications.stream().filter(Objects::nonNull).forEach(notification -> notificationService.notify(notification));
+        }
+    }
+
+    public boolean isNotificationFeatureAvailable () {
+        return notificationService != null;
+    }
+
+    private boolean enableScheduledNotifications() {
+        return isNotificationFeatureAvailable() && this.notificationTasks != null && !this.notificationTasks.isEmpty();
```

Comment:
No need to start scheduler thread if no NotificationTasks were provided. The NotificationManager will still be available for sending ad-hoc notifications.